test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,390 @@
+		customDnsService.Name, metav1.GetOptions{})
+	require.NoError(t, err, "Error when getting custom DNS service object : %v", err)
+
+	if customCoreDnsServiceObject.Spec.ClusterIP == "" {
+		require.Fail(t, "ClusterIP is empty for the custom DNS service")
+	}
+
+	t.Logf("ClusterIP of the dns service: %s\n", customCoreDnsServiceObject.Spec.ClusterIP)
+	return customCoreDnsServiceObject.Spec.ClusterIP
+}
+
+func getIPFamily() *corev1.IPFamily {
+	ipFamily := corev1.IPv4Protocol
+	return &ipFamily
+}
+
+func setDnsServerAddressInAntrea(t *testing.T, data *TestData, dnsServiceIP string) {
+	var agentConf agentconfig.AgentConfig
+	cm, err := data.GetAntreaConfigMap(antreaNamespace)
+	require.NoError(t, err, "Error when getting antrea configMap : %v", err)
+
+	err = yaml.Unmarshal([]byte(cm.Data["antrea-agent.conf"]), &agentConf)
+	require.NoError(t, err, "failed to unmarshal antrea agent config from ConfigMap: %v", err)
+
+	agentChanges := func(config *agentconfig.AgentConfig) {
+		config.DNSServerOverride = dnsServiceIP
+	}
+	err = data.mutateAntreaConfigMap(nil, agentChanges, false, true)
+	require.NoError(t, err, "Error when setting up customDNS server IP in Antrea configmap : %v", err)
+
+	t.Logf("dns server value set to %+v in antrea \n", dnsServiceIP)
+
+}
+
+func createDnsConfig(ipForConfig string) map[string]string {
+	config := fmt.Sprintf(`lfx.test:53 {
+    errors
+    health
+    hosts {
+        %s %s
+        no_reverse
+        pods verified
+        ttl 10
+    }
+    loop
+    reload
+}`, ipForConfig, testFullyQualifiedDomainName)
+
+	configData := map[string]string{
+		"Corefile": config,
+	}
+
+	return configData
+}
+
+func createDnsPod(t *testing.T, data *TestData) {
+	volume := []corev1.Volume{
+		{
+			Name: customDnsVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: customDnsConfigName,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "Corefile",
+							Path: "Corefile",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	volumeMount := []corev1.VolumeMount{
+		{
+			Name:      customDnsVolume,
+			MountPath: "/etc/coredns",
+		},
+	}
+
+	label := map[string]string{customDnsLabelKey: customDnsLabelValue}
+	pb := NewPodBuilder(customDnsPodName, data.testNamespace, customDnsImage)
+	pb.WithLabels(label)
+	pb.WithAnnotations(map[string]string{"Foo": "Bar"})
+	pb.WithContainerName(customDnsContainerName)
+	pb.WithArgs([]string{"-conf", "/etc/coredns/Corefile"})
+	pb.AddVolume(volume)
+	pb.AddVolumeMount(volumeMount)
+
+	require.NoError(t, pb.Create(data))
+	_, err := data.podWaitForIPs(defaultTimeout, customDnsPodName, data.testNamespace)
+	require.NoError(t, err)
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, customDnsPodName, data.testNamespace))
+
+}
+
+func buildFqdnPolicy(t *testing.T) *utils.ClusterNetworkPolicySpecBuilder {
```

Comment:
could we use an ANP instead of an ACNP?
overall it makes cleanup easier, as the ANP can belong to the test Namespace, which means it is deleted when the test Namespace is automatically deleted at the end of the test case